test/test-reporter.mjs

```diff
@@ -0,0 +1,66 @@
+function colorize (type, text) {
+  if (type === 'pass') {
+    const whiteText = `\x1b[30m${text}`
+    // Green background with black text
+    return `\x1b[42m${whiteText}\x1b[0m`
+  }
+
+  if (type === 'fail') {
+    const blackText = `\x1b[37m${text}`
+    // Red background with white text
+    return `\x1b[41m${blackText}\x1b[0m`
+  }
+
+  return text
+}
+
+function formatDiagnosticStr (str) {
+  return str.replace(/^(\w+)(\s*\d*)/i, (_, firstWord, rest) => {
+    return firstWord.charAt(0).toUpperCase() + firstWord.slice(1).toLowerCase() + ':' + rest
+  })
+}
+
+async function * reporter (source) {
+  const failed = new Set()
+  const diagnostics = new Set()
+  diagnostics.add('\n\n')
```

Comment:
Let's leave formatting of output to the display section of the code and only display diagnostic messages when any exist (as we do with failed files).